test/preview/components/rankers/test_similarity.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+
+from haystack.preview import Document, ComponentError
+from haystack.preview.components.rankers.similarity import SimilarityRanker
+
+
+class TestSimilarityRanker:
+    @pytest.mark.unit
+    def test_to_dict(self):
+        component = SimilarityRanker(model_name_or_path="cross-encoder/ms-marco-MiniLM-L-6-v2")
+        data = component.to_dict()
+        assert data == {
+            "type": "SimilarityRanker",
+            "init_parameters": {"device": "cpu", "model_name_or_path": "cross-encoder/ms-marco-MiniLM-L-6-v2"},
+        }
+
+    @pytest.mark.unit
+    def test_to_dict_with_custom_init_parameters(self):
+        component = SimilarityRanker()
+        data = component.to_dict()
+        assert data == {
+            "type": "SimilarityRanker",
+            "init_parameters": {"device": "cpu", "model_name_or_path": "cross-encoder/ms-marco-MiniLM-L-6-v2"},
+        }
+
+    @pytest.mark.integration
+    def test_from_dict(self):
+        data = {
+            "type": "SimilarityRanker",
+            "init_parameters": {"device": "cpu", "model_name_or_path": "cross-encoder/ms-marco-MiniLM-L-6-v2"},
+        }
+        component = SimilarityRanker.from_dict(data)
+        assert component.model_name_or_path == "cross-encoder/ms-marco-MiniLM-L-6-v2"
+
+    @pytest.mark.integration
+    def test_run(self):
+        """
+        Test if the component runs correctly.
+        """
+        sampler = SimilarityRanker(model_name_or_path="cross-encoder/ms-marco-MiniLM-L-6-v2")
+        sampler.warm_up()
+        docs = [Document(text="Berlin"), Document(text="Belgrade"), Document(text="Sarajevo")]
+        query = "City in Bosnia and Herzegovina"
+        output = sampler.run(query=query, documents=docs)
+        docs = output["documents"]
+        assert len(docs) == 3
+        assert docs[0].text == "Sarajevo"
```

Comment:
I think it would be good to also add an assert checking that the scores are in descending order. So something like 
```python
assert sorted([d.score for d in docs], reverse=True) == [d.score for d in docs]
```
What do you think?

Reply:
Awesome suggestion @sjrl caught a subtle bug, see [82bc1de](https://github.com/deepset-ai/haystack/pull/5923/commits/82bc1de53434b250ef640e49491a405db2156a97)